utilities/measurement_utility.cc

```diff
@@ -15,95 +15,141 @@
 
 // make_measurement.exe --type=hash --input=input-file --output=output-file
 
+#include <sstream>
 #include <gflags/gflags.h>
 #include "certifier.h"
 #include "support.h"
 
+using namespace std;
 using namespace certifier::utilities;
 
+int parse_other_files_size(string &        other_files,
+                           vector<string> &other_files_list,
+                           vector<int> &   other_files_size);
+
 DEFINE_bool(print_all, false, "verbose");
 DEFINE_bool(print_debug, false, "print debugging info");
 DEFINE_string(type, "hash", "measurement type");
 DEFINE_string(input, "measurement_utility.exe", "input file");
-DEFINE_string(policy_key, "", "policy key file, e.g., policy_key.py");
+DEFINE_string(other_files,
+              "",
+              "Comma-separated list of other files to include in measurement; "
+              "e.g., policy_key.py,certifier_framework.py");
 DEFINE_string(output, "measurement_utility.exe.measurement", "output file");
 
 
 const int sha256_size = 32;
 
-int hash_utility(string &input, string &policy_key_file, string &output) {
+/*
+ * Compute the hash-digest for the input file and, optionally, include
+ * a list of other-files that need to be included in the measurement,
+ * specified by the other_files argument.
+ *
+ * NOTE: We make sure to read-in the contents of all files that go into the
+ * measurement ino one buffer. This slightly reduces the chance that while
+ * this utility is running, we don't run into the Time-of-Check to Time-of-Use,
+ * abbreviated as "TOCTOU", 'vulnerability'. That might happen if we read-in
+ * file-1 first, and while other files are being read, contents of file-1
+ * is changed. Second, and more importantly, we use the digest_message()
```

Comment:
If my assessment with the newly added test case is correct, then the 2nd point starting from "Second ..." is valid.

Otherwise, I'll have to withdraw the test case and rework this comment.

Please review this carefully and let's settle what the right thing to do is.